src/core/ext/xds/xds_api.cc

```diff
@@ -3069,6 +3071,10 @@ grpc_error_handle CdsResponseParse(
             resource_names_failed->insert(cluster_name);
             continue;
           }
+        } else {
```

Comment:
Same comment here.  Let's handle the error case right next the condition that causes it, so that we can de-indent the bulk of the code here.